appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/WebServiceInvocationHandler.java

```diff
@@ -129,96 +136,76 @@ public Object invoke(Object proxy, Method method, Object[] args)
+                    if (!authorized) {
+                        throw new AccessLocalException("Client not authorized to access " + inv.method);
+                    }
+                }
+            } else if (hasHandlers_) {
+
+                // jaxws enpoint
+                // authorization was done in security pipe
+                // Now that application handlers have run, do
+                // another method lookup and compare the results
+                // with the original one. This ensures that the
                 // application handlers have not changed
                 // which method is invoked.
 
-                Method methodBefore = inv.getWebServiceMethod();
-                
+                final Method methodBefore = inv.getWebServiceMethod();
+
                 if (methodBefore != null && !methodBefore.equals(inv.method)) {
-                    inv.exception = new UnmarshalException
-			    (localStrings.getLocalString
-			    ("enterprise.webservice.postHandlerMethodMismatch",
-			    "Original Method {0} does not match post-handler method {1}",
-			    new Object[] { methodBefore, inv.method }));
-		        throw inv.exception;
-		        }
+                    inv.exception = new UnmarshalException(
+                        localStrings.getLocalString("enterprise.webservice.postHandlerMethodMismatch",
+                            "Original Method {0} does not match post-handler method {1}",
+                            new Object[] {methodBefore, inv.method}));
+                    throw inv.exception;
+                }
             }
 
-
-            ComponentContext ctx = container.getContext(inv);
-            inv.context  = ctx;
-            inv.ejb      = ctx.getEJB();
+            final ComponentContext ctx = container.getContext(inv);
+            inv.context = ctx;
+            inv.ejb = ctx.getEJB();
             inv.instance = inv.ejb;
 
             container.preInvokeTx(inv);
 
             // Enterprise Bean class doesn't necessarily implement
             // web service endpoint interface, so we can't directly
             // dispatch through the given method object.
-            Method beanClassMethod = ejbClass_.getMethod
-                (method.getName(), method.getParameterTypes());
+            final Method beanClassMethod = ejbClass_.getMethod(method.getName(), method.getParameterTypes());
             inv.beanMethod = beanClassMethod;
             inv.methodParams = args;
-            returnValue = container.intercept(inv);
-        } catch(NoSuchMethodException nsme) {
+            return container.intercept(inv);
+        } catch (NoSuchMethodException nsme) {
             inv.exception = nsme;
-        } catch(InvocationTargetException ite) {
+            return null;
+        } catch (InvocationTargetException ite) {
             inv.exception = ite.getCause();
-        } catch(Throwable c) {
+            return null;
+        } catch (Throwable c) {
```

Comment:
Catching Throwable is a bad idea. Always. Even in application servers. Maybe we should give it some attention in Payara 6.